cdap-notifications/src/main/java/co/cask/cdap/notifications/service/NotificationService.java

```diff
@@ -0,0 +1,99 @@
+   * @param notification notification object to send.
+   * @param <N> Type of the notification to send.
+   * @return a {@link ListenableFuture} describing the state of the async send operation.
+   * @throws NotificationException in case of any error when publishing the notification.
+   * @throws NotificationFeedException in case of any error regarding the {@code feed}.
+   */
+  <N> ListenableFuture<N> publish(NotificationFeed feed, N notification)
+    throws NotificationException, NotificationFeedException;
+
+  /**
+   * Send one Notification asynchronously. The {@code notificationType} is used to serialize the notification
+   * passed to the Notification system.
+   *
+   * @param feed {@link NotificationFeed} where to publish the notification.
+   * @param notification notification object to send.
+   * @param notificationType type to use to serialize the notification in the Notification system.
+   * @param <N> Type of the notification to send.
+   * @return a {@link ListenableFuture} describing the state of the async send operation.
+   * @throws NotificationException in case of any error when publishing the notification.
+   * @throws NotificationFeedException in case of any error regarding the {@code feed}.
+   */
+  <N> ListenableFuture<N> publish(NotificationFeed feed, N notification, Type notificationType)
+    throws NotificationException, NotificationFeedException;
+
+  /**
+   * Subscribe to the notification received on the {@code feed}, and handle the notifications with the {@code handler}.
+   * Before this call is made, the {@code feed} has to be created using the
+   * {@link NotificationFeedManager}.
+   * Multiple subscriptions to a same feed with different handlers are possible.
+   *
+   * @param feed {@link NotificationFeed} to subscribe to.
+   * @param handler {@link NotificationHandler} that will handle the notifications coming from the feed.
+   * @param <N> Type of the notifications.
+   * @return A {@link Cancellable} for cancelling Notification consumption.
+   * @throws NotificationFeedNotFoundException if the feed does not exist, according to the
+   * {@link NotificationFeedManager}.
+   * @throws NotificationFeedException in case of any other error concerning the feed.
+   */
+  <N> Cancellable subscribe(NotificationFeed feed, NotificationHandler<N> handler)
+    throws NotificationFeedException;
+
+  /**
+   * Subscribe to the notification received on the {@code feed}, and handle the notifications with the {@code handler}.
+   * Before this call is made, the {@code feed} has to be created using the
+   * {@link NotificationFeedManager}.
+   * Multiple subscriptions to a same feed with different handlers are possible.
+   *
+   * @param feed {@link NotificationFeed} to subscribe to.
+   * @param handler {@link NotificationHandler} that will handle the notifications coming from the feed.
+   * @param executor {@link Executor} to use to perform the polling/pushing of notifications from the Notification
```

Comment:
That is true too. Right now the executor is only used in the in-memory implementation of the ``NotificationService``, and it is used to push the notifications received by the system to all the subscribed handlers. So it does those two things.